src/arithmetic.jl

```diff
@@ -407,89 +365,70 @@ function /{T<:Integer, S<:NumberNotSeries}(a::Taylor1{Rational{T}}, b::S)
 end
 
 for T in (:Taylor1, :HomogeneousPolynomial, :TaylorN)
-    @eval begin
-        /{T<:NumberNotSeries}(a::$T{T}, b::T) = a * inv(b)
-        function /{T<:NumberNotSeries,S<:NumberNotSeries}(a::$T{T}, b::S)
-            R = promote_type(T,S)
-            return convert($T{R}, a) * inv(convert(R, b))
-        end
-        /{T<:NumberNotSeries}(a::$T, b::T) = a * inv(b)
+
+    @eval /{T<:NumberNotSeries}(a::$T{T}, b::T) = a * inv(b)
+
+    @eval function /{T<:NumberNotSeries,S<:NumberNotSeries}(a::$T{T}, b::S)
+        R = promote_type(T,S)
+        return convert($T{R}, a) * inv(convert(R, b))
     end
+
+    @eval /{T<:NumberNotSeries}(a::$T, b::T) = a * inv(b)
 end
 
 for T in (:HomogeneousPolynomial, :TaylorN)
-    @eval begin
-        function /{T<:NumberNotSeries,S<:NumberNotSeries}(
-                b::$T{Taylor1{S}}, a::Taylor1{T})
-            @inbounds aux = b[1] / a
-            R = typeof(aux)
-            coeffs = Array{R}(length(b.coeffs))
-            coeffs .= b.coeffs ./ a
-            return $T(coeffs, b.order)
-        end
-        function /{T<:NumberNotSeries,S<:NumberNotSeries}(
-                b::Taylor1{$T{S}}, a::$T{T})
-            @inbounds aux = b[1] / a
-            R = typeof(aux)
-            coeffs = Array{R}(length(b.coeffs))
-            coeffs .= b.coeffs ./ a
-            return Taylor1(coeffs, b.order)
-        end
+    @eval function /{T<:NumberNotSeries,S<:NumberNotSeries}(
+            b::$T{Taylor1{S}}, a::Taylor1{T})
+        @inbounds aux = b[1] / a
+        R = typeof(aux)
+        coeffs = Array{R}(length(b.coeffs))
+        coeffs .= b.coeffs ./ a
+        return $T(coeffs, b.order)
     end
-end
 
+    @eval function /{T<:NumberNotSeries,S<:NumberNotSeries}(
```

Comment:
I think both functions are needed: the first function is e.g. for `\(b::TaylorN{Taylor1{T}}, a::Taylor1{S})`, while the second one is for `\(b::Taylor1{TaylorN{T}}, a::TaylorN{S})`. Note that the code in 376 does not include this case.

In any case, you have a point: the body of the functions are the same, while it is only the function definition what changes. I was simply too lazy to think on collapsing this further.